### PR TITLE
Add CLUE Benchmark (11 datasets)

### PR DESCRIPTION
**Pull request by JetRunner**
Add 11 tasks of [CLUE](https://github.com/CLUEbenchmark/CLUE).

### CODE REVIEW

File: datasets/clue/clue.py [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+            data_url="https://storage.googleapis.com/cluebenchmark/tasks/drcd_public.zip",
+            url="https://github.com/DRCKnowledgeTeam/DRCD",
+        ),
+        ClueConfig(
+            name="chid",
+            description=textwrap.dedent(
+                """\
+            Delta Reading Comprehension Dataset (DRCD) belongs to the general field of traditional
+            Chinese machine reading comprehension data set. This data set is expected to become a
+            standard Chinese reading comprehension data set suitable for transfer learning.
```
Comment (lhoestq) [inline datasets/clue/clue.py:230]: This is the description of `dcrd`, not `chid` ^^
Maybe a copy/paste issue ?

File: datasets/clue/clue.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+                   year={2019}
+                }"""),
+        ),
+        ClueConfig(
+            name="c3",
+            description=textwrap.dedent(
+                """\
+            Delta Reading Comprehension Dataset (DRCD) belongs to the general field of traditional
+            Chinese machine reading comprehension data set. This data set is expected to become a
+            standard Chinese reading comprehension data set suitable for transfer learning.
```
Comment (lhoestq) [inline datasets/clue/clue.py:254]: same here

File: datasets/clue/clue.py [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+            citation=self.config.citation + "\n" + _CLUE_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        dl_dir = dl_manager.download_and_extract(self.config.data_url)
+        data_dir = os.path.join(dl_dir, self.config.data_dir)
+        test_split = nlp.SplitGenerator(
+            name=nlp.Split.TEST,
+            gen_kwargs={
+                "data_file": os.path.join(data_dir or "", "test.json" if self.config.name != "diagnostics" else "diagnostics_test.json"),
```
Comment (lhoestq) [inline datasets/clue/clue.py:365]: Maybe replace `data_dir or ""` by `data_dir`, as `data_dir` is never None ?

File: datasets/clue/clue.py [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+
+                        elif self.config.label_column in row:
+                            label = row[self.config.label_column]
+                            # Notice: some labels in CMNLI are missing. We drop these data.
+                            if self.config.name == 'cmnli' and label == '-':
+                                continue
+                            # For some tasks, the label is represented as 0 and 1 in the tsv
+                            # files and needs to be cast to integer to work with the feature.
+                            if label_classes and label not in label_classes:
+                                print(label, label_classes)
```
Comment (lhoestq) [inline datasets/clue/clue.py:475]: Could you remove this `print` statement ?

**Review by lhoestq — COMMENTED** [review]
This is awesome ! Thank you so much for adding this one :)

Maybe we can use `ClassLabel` feature type for labels, as in `glue` ? What do you think ?

I also left a few (minor) comments:

**Comment by JetRunner** [comment]
Thanks, @lhoestq! I've addressed the comments. 
Also, I have tried to use `ClassLabel` [when possible](https://github.com/huggingface/nlp/pull/572/files#diff-1026ac7d7b78bf029cb0ebe63162c77dR297). Is there still somewhere else we can use `ClassLabel`?

**Comment by JetRunner** [comment]
I believe CI failure is unrelated.

**Comment by brightmart** [comment]
Great job!

**Review by lhoestq — APPROVED** [review]
Looks good to me, thanks !
Don't pay attention to the CI error, it comes from master and we're currently trying to fix it

**Pull request merged**